packages/material-ui-unstyled/src/ButtonUnstyled/ButtonUnstyled.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_useForkRef as useForkRef } from '@material-ui/utils';
+import composeClasses from '../composeClasses';
+import { getButtonUnstyledUtilityClass } from './buttonUnstyledClasses';
+import ButtonUnstyledProps, {
+  ButtonUnstyledOwnProps,
+  ButtonUnstyledTypeMap,
+} from './ButtonUnstyledProps';
+import useButton from './useButton';
+import appendOwnerState from '../utils/appendOwnerState';
+
+export interface ButtonUnstyledOwnerState extends ButtonUnstyledOwnProps {
+  focusVisible: boolean;
+  active: boolean;
+}
+
+const useUtilityClasses = (ownerState: ButtonUnstyledOwnerState) => {
+  const { active, disabled, focusVisible } = ownerState;
+
+  const slots = {
+    root: ['root', disabled && 'disabled', focusVisible && 'focusVisible', active && 'active'],
+  };
+
+  return composeClasses(slots, getButtonUnstyledUtilityClass, {});
+};
+/**
+ * The foundation for building custom-styled buttons.
+ *
+ * Demos:
+ *
+ * - [Buttons](https://material-ui.com/components/buttons/)
+ *
+ * API:
+ *
+ * - [ButtonUnstyled API](https://material-ui.com/api/button-unstyled/)
+ */
+const ButtonUnstyled = React.forwardRef(function ButtonUnstyled<
+  D extends React.ElementType = ButtonUnstyledTypeMap['defaultComponent'],
+>(props: ButtonUnstyledProps<D>, ref: React.ForwardedRef<any>) {
+  const {
+    className,
+    component,
+    components = {},
+    componentsProps = {},
+    children,
+    disabled,
+    action,
+    onBlur,
+    onClick,
+    onFocus,
+    onFocusVisible,
+    onKeyDown,
+    onKeyUp,
+    onMouseLeave,
+    ...other
+  } = props;
+
+  const buttonRef = React.useRef<HTMLButtonElement | HTMLAnchorElement | HTMLElement>();
+  const handleRef = useForkRef(buttonRef, ref);
+
+  const { active, focusVisible, setFocusVisible, getRootProps } = useButton({
+    ...props,
+    ref: handleRef,
+  });
+
+  React.useImperativeHandle(
+    action,
+    () => ({
+      focusVisible: () => {
+        setFocusVisible(true);
+        buttonRef?.current?.focus();
```

Comment:
I can't remember the last time I saw a value usage of ? with React ref. Here, the first one seems unnecessary since it's coming from a React.useRef, the second one seems only unnecessary since refs resolves before imperative handle. Would it make more sense with?

```suggestion
        buttonRef.current!.focus();
```